pkg/graph/resolvers/key_contact_category_test.go

```diff
@@ -0,0 +1,421 @@
+
+	testCases := []struct {
+		name      string
+		category  string
+		expected  string // Expected stored value (empty string means should error)
+		shouldErr bool
+	}{
+		// Whitespace-only (should error)
+		{"whitespace only - spaces", "   ", "", true},
+		{"whitespace only - tabs", "\t\t\t", "", true},
+		{"whitespace only - newlines", "\n\n\n", "", true},
+		{"whitespace only - mixed", " \t\n ", "", true},
+		// Whitespace with content (should sanitize)
+		{"leading space", "  Updated Category", "Updated Category", false},
+		{"trailing space", "Updated Category  ", "Updated Category", false},
+		{"leading and trailing", "  Updated Category  ", "Updated Category", false},
+		{"leading tab", "\tUpdated Category", "Updated Category", false},
+		{"trailing newline", "Updated Category\n", "Updated Category", false},
+		{"newline in middle", "Updated\nCategory", "UpdatedCategory", false},
+	}
+
+	for _, tc := range testCases {
+		suite.Run(tc.name, func() {
+			name := tc.category
+
+			updatedCategory, err := UpdateKeyContactCategory(
+				suite.testConfigs.Context,
+				suite.testConfigs.Logger,
+				suite.testConfigs.Principal,
+				suite.testConfigs.Store,
+				category.ID,
+				name,
+			)
+
+			if tc.shouldErr {
+				suite.Error(err)
+				suite.Contains(err.Error(), "category name cannot be empty or whitespace-only")
+			} else {
+				suite.NoError(err)
+				suite.NotNil(updatedCategory)
+				suite.Equal(tc.expected, updatedCategory.Name)
+			}
+		})
+	}
+}
+
+// TestUpdateKeyContactCategory tests updating an existing key contact category.
+func (suite *ResolverSuite) TestUpdateKeyContactCategory() {
+	originalCategoryName := "Original Category"
+	updatedCategoryName := "Updated Category"
+
+	// Create a category first
+	category := suite.createKeyContactCategory(originalCategoryName)
+
+	// Update the category
+	name := updatedCategoryName
+
+	updatedCategory, err := UpdateKeyContactCategory(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		category.ID,
+		name,
+	)
+
+	suite.NoError(err)
+	suite.NotNil(updatedCategory)
+	suite.Equal(category.ID, updatedCategory.ID)
+	suite.Equal(updatedCategoryName, updatedCategory.Name)
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, updatedCategory.CreatedBy)
+	suite.NotNil(updatedCategory.ModifiedBy)
+	suite.Equal(suite.testConfigs.Principal.UserAccount.ID, *updatedCategory.ModifiedBy)
+	suite.NotNil(updatedCategory.ModifiedDts)
+}
+
+// TestUpdateKeyContactCategory_NotFound tests updating a non-existent category.
+func (suite *ResolverSuite) TestUpdateKeyContactCategory_NotFound() {
+	nonExistentID := uuid.New()
+	name := "Updated Category"
+
+	_, err := UpdateKeyContactCategory(
+		suite.testConfigs.Context,
+		suite.testConfigs.Logger,
+		suite.testConfigs.Principal,
+		suite.testConfigs.Store,
+		nonExistentID,
+		name,
+	)
+
+	suite.Error(err)
+	suite.Contains(err.Error(), "record not found for given key")
```

Comment:
```suggestion
 suite.True(errors.Is(err, loaders.ErrRecordNotFoundForKey))
```

This is a typed error, so go has functionality to compare it by type instead of by comparing strings.